storagemarket/lp2pimpl/net.go

```diff
@@ -325,12 +325,23 @@ func (p *DealProvider) getDealStatus(req types.DealStatusRequest) types.DealStat
 		return errResp("getting sector status from sealer")
 	}
 
+	sealingStatus := string(si.State)
+	var found bool
+
+	if storagemarket.IsFinalSealingState(si.State) {
+		found = storagemarket.HasDeal(si.Deals, pds.ChainDealID)
+	}
+
+	if !found {
+		sealingStatus = storagemarket.ErrDealNotFound.Error()
+	}
+
```

Comment:
If the deal has not reached the proving state, it looks like this code will always set the SealingStatus to ErrDealNotFound